rust/src/csv/writer.rs

```diff
@@ -0,0 +1,139 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::fs::File;
+use std::io::{BufWriter, Write};
+
+use array::*;
+use datatypes::DataType;
+use record_batch::RecordBatch;
+
+pub struct Writer {
+    w: BufWriter<File>,
+}
+
+macro_rules! write_primitive_array {
+    ($WRITER:expr, $BATCH:expr, $ROW_INDEX:expr, $COL_INDEX:expr, $TY:ty) => {{
+        let array = $BATCH
+            .column($COL_INDEX)
+            .as_any()
+            .downcast_ref::<$TY>()
+            .unwrap();
+        $WRITER
+            .write(format!("{}", array.value($ROW_INDEX)).as_bytes())
+            .unwrap();
+    }};
+}
+
+impl Writer {
+    pub fn new(file: File) -> Self {
+        Writer {
+            w: BufWriter::new(file),
+        }
+    }
+
+    pub fn write(&mut self, batch: &RecordBatch) {
+        for row_index in 0..batch.num_rows() {
+            for col_index in 0..batch.num_columns() {
+                if col_index > 0 {
+                    self.w.write(",".as_bytes()).unwrap();
```

Comment:
seems we are ignoring possible errors in many places. Should we surface these errors instead of `unwrap`?